File: 6. ZigZag Conversion/Solution.py
class Solution:
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows == 1:
            return s
        n = len(s)
        ans = str()
        increase = 2 * numRows - 2
        for i in range(numRows):
            j = 0
            while i + j < n:
                ans += s[i + j]
                if i != 0 and i != numRows - 1 and j + increase - i < n:
                    ans += s[j + increase - i]
                j += increase
        return ans

File: 6. ZigZag Conversion/Solution.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1)
            return s;
        string ans;
        int n = s.size();
        int increase = 2 * numRows - 2;
        for (int i = 0; i < numRows; i++) {
            for (int j = 0; i + j < n; j += increase) {
                ans += s[i + j];
                if (i != 0 && i != numRows - 1 && j + increase - i < n)
                    ans += s[j + increase - i];
            }
        }
        return ans;
    }
};